x/bep3/legacy/v0_11/types.go

```diff
@@ -0,0 +1,81 @@
+package v0_11
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	tmbytes "github.com/tendermint/tendermint/libs/bytes"
+)
+
+// GenesisState - all bep3 state that must be provided at genesis
+type GenesisState struct {
+	Params      Params      `json:"params" yaml:"params"`
+	AtomicSwaps AtomicSwaps `json:"atomic_swaps" yaml:"atomic_swaps"`
+}
+
+// Params governance parameters for the bep3 module
+type Params struct {
+	AssetParams AssetParams `json:"asset_params" yaml:"asset_params"`
+}
+
+// AssetParam parameters that must be specified for each bep3 asset
+type AssetParam struct {
+	Denom         string         `json:"denom" yaml:"denom"`                                     // name of the asset
+	CoinID        int            `json:"coin_id" yaml:"coin_id"`                                 // SLIP-0044 registered coin type - see https://github.com/satoshilabs/slips/blob/master/slip-0044.md
+	SupplyLimit   AssetSupply    `json:"supply_limit" yaml:"supply_limit"`                       // asset supply limit
+	Active        bool           `json:"active" yaml:"active"`                                   // denotes if asset is available or paused
+	DeputyAddress sdk.AccAddress `json:"deputy_address" yaml:"deputy_address"`                   // the address of the relayer process
+	FixedFee      sdk.Int        `json:"incoming_swap_fixed_fee" yaml:"incoming_swap_fixed_fee"` // the fixed fee charged by the relayer process for incoming swaps
+	MinSwapAmount sdk.Int        `json:"min_swap_amount" yaml:"min_swap_amount"`                 // Minimum swap amount
+	MaxSwapAmount sdk.Int        `json:"max_swap_amount" yaml:"max_swap_amount"`                 // Maximum swap amount
+	MinBlockLock  uint64         `json:"min_block_lock" yaml:"min_block_lock"`                   // Minimum swap block lock
+	MaxBlockLock  uint64         `json:"max_block_lock" yaml:"max_block_lock"`                   // Maximum swap block lock
+}
+
+// AssetParams array of AssetParam
+type AssetParams []AssetParam
+
+// AssetSupply contains information about an asset's supply
+type AssetSupply struct {
+	IncomingSupply sdk.Coin `json:"incoming_supply"  yaml:"incoming_supply"`
+	OutgoingSupply sdk.Coin `json:"outgoing_supply"  yaml:"outgoing_supply"`
+	CurrentSupply  sdk.Coin `json:"current_supply"  yaml:"current_supply"`
+	SupplyLimit    sdk.Coin `json:"supply_limit"  yaml:"supply_limit"`
```

Comment:
The AssetSupply in types/params.go doesn't have the SupplyLimit field, is it meant to match here?

Reply:
corrected in #623